src/plugins/npm.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2018-present Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain
+ * a copy of the license in the LICENSE and PATENTS files in the root of this
+ * repository.
+ */
+
+import { spawn } from "child_process";
+import { ICompiler, IFile, INpmPackage, INpmPluginData, IPlugin } from "../client";
+
+/**
+ * The `NpmPlugin` parses `package.json` files and emits data about each
+ * NPM package. Packages marked `private: true` will be ignored.
+ */
+export class NpmPlugin implements IPlugin<INpmPluginData> {
+    public async compile(packageJsons: IFile[], dm: ICompiler): Promise<INpmPluginData> {
+        const info = await Promise.all(packageJsons.map(pkg => this.parseNpmInfo(pkg, dm)));
+        const npm = arrayToObject(info.filter(isDefined), pkg => pkg.name);
+        return { npm };
+    }
+
+    private parseNpmInfo = async (packageJson: IFile, dm: ICompiler): Promise<INpmPackage | undefined> => {
+        const json = JSON.parse(packageJson.read());
+        if (json.private === true) {
+            // ignore private packages as they will not appear in `npm info`
+            return undefined;
+        }
+        const data = JSON.parse(await this.getNpmInfo(json.name));
+        return {
+            name: data.name,
+            // tslint:disable-next-line:object-literal-sort-keys
+            description: data.description,
+            latestVersion: data["dist-tags"].latest,
```

Comment:
is this guaranteed to be non-null? what about not-yet released libraries?

Reply:
yes good catch. un-released libraries will actually die here because `npm info` will return nothing